src/main/java/seedu/address/logic/parser/TimeslotParser.java

```diff
@@ -0,0 +1,261 @@
+            formatter.append("dd/MM/yyyy ");
+        } else if (dateInput.matches(TimeslotRegex.DATE_DASH_SHORT)) {
+            formatter.append("dd-MM-yy ");
+        } else {
+            formatter.append("dd-MM-yyyy ");
+        }
+        return parseTimeFormat(timeInput, formatter);
+    }
+
+    /**
+     * Parses a {@code String timeInput, StringBuilder} into a full {@code StringBuilder} format for
+     * LocalDateTime. 24-Hour Clock and Meridian time Inputs are both accepted.
+     */
+    public static StringBuilder parseTimeFormat(String timeInput, StringBuilder formatter) {
+        if (timeInput.contains("AM") || timeInput.contains("PM")) {
+            formatter.append("hh:mma");
+        } else {
+            formatter.append("HH:mm");
+        }
+        return formatter;
+    }
+
+    /**
+     * Parses a {@code String userInput} into a {@code LocalDateTime}.
+     * Parses user input containing the next day, month or year. Time can remain the same or
+     * revised into a new one as per user input.
+     *
+     * @throws ParseException if the given {@code dateTime} does not
+     * conform to the expected date time format.
+     * @throws NullPointerException if the {@code timeInput} does not
+     * conform to the expected time format.
+     */
+    public static LocalDateTime parseNextDateTime(String userInput) throws ParseException, NullPointerException {
+        try {
+            LocalDateTime currentDateTime = LocalDateTime.now().withSecond(0).withNano(0);
+            String[] nextDateTimeInputArray = userInput.split(REMOVE_WHITESPACE_REGEX);
+            String keyword = nextDateTimeInputArray[1];
+            LocalDateTime parsedDate = null;
+
+            if (Arrays.stream(Day.values()).anyMatch(e -> e.name().equals(keyword))) {
+                parsedDate = currentDateTime.with(TemporalAdjusters.next(DayOfWeek.valueOf(keyword)));
+            } else if (keyword.contains("MONTH")) {
+                parsedDate = currentDateTime.plusMonths(1);
+            } else if (keyword.contains("YEAR")) {
+                parsedDate = currentDateTime.plusYears(1);
+            } else {
+                throw new ParseException(MESSAGE_INVALID_TIME_FORMAT);
+            }
+
+            if (nextDateTimeInputArray.length > 2) {
+                String timeInput = nextDateTimeInputArray[2];
+                int[] hoursMinutesArray = parseTime(timeInput);
+                parsedDate = parsedDate.withHour(hoursMinutesArray[0]);
+                parsedDate = parsedDate.withMinute(hoursMinutesArray[1]);
+                parsedDate = parsedDate.withSecond(0).withNano(0);
+            }
+
+            return parsedDate;
+
+        } catch (DateTimeParseException e) {
+            throw new ParseException(MESSAGE_INVALID_NEXT_DATE_TIME_FORMAT);
+        } catch (NullPointerException e) {
+            throw new NullPointerException(MESSAGE_INVALID_TIME_FORMAT);
+        }
```

Comment:
a `NullPointerException` shouldn't be thrown, because the caller method don't handle it. should throw `ParseException` instead. also `NullPointerException`s technically shouldn't be thrown/caught either since they're runtime exceptions, not checked exceptions.

Reply:
Ahh okay, I didn't know about the runtime and checked exceptions part. Will revise it, thanks!